### DISCUSSION TITLE
Issues when connecting dapp and wallet by deeplink

### DISCUSSION THREAD

**Discussion by cp-byungsun** [WalletConnect/walletconnect-monorepo#2312] [Developer Support]
WalletConnectV2 
When calling the wallet with a deep link, when the wallet approves the connection request from the Dapp, the Dapp is in the background and cannot send a response to the wallet. So a timeout occurs in the wallet. However, Dapp and wallet are connected.
Dapp & wallet are native android. 

How can I solve this issue?

**Comment by finessevanes** [comment]
@cp-byungsun can you provide us a little more information such as what SDK you're using and the code implementation?

**Reply by cp-byungsun** [reply]
using sdk com.walletconnect:android-bom:1.9.1

dapp side 
`  val methods = listOf(
                "eth_sendTransaction",
                "eth_signTransaction",
                "eth_sign",
                "personal_sign",
                "eth_signTypedData",
                "get_balance"
            )
            val events = listOf("accountsChanged", "chainChanged", "connect", "disconnect")
            val namespaces =
                mapOf("eip155" to Sign.Model.Namespace.Proposal(chains, methods, events))

            // for proposer to create a session
            val connectParams = Sign.Params.Connect(namespaces, null, null, pairing!!)

            SignClient.connect(connectParams,
                onSuccess = { -> /*Session proposal was being sent successfully over pairing topic*/
                    Timber.tag("SignClient_Connect_Proposal_Send_Success")
                        .d("The sign client connection proposal was successfully sent to the wallet")
                    Timber.tag("SignClient_Connect_Proposal_Send_Success")
                        .d("connectParam = " + connectParams.namespaces)
                },
                onError = { error -> /*callback for error while sending session proposal*/
                    Timber.tag("CONNECTION_ERROR").e(error.throwable.stackTraceToString())
                })`
 and call deeplink 

wallet side get request approve session and confirm 
but timeout occurred in wallet confirm response

**Reply by finessevanes** [reply]
What wallet are you using when testing?

**Reply by cp-byungsun** [reply]
This is a wallet in development. It is a wemix wallet and is not registered in walletconnect.

**Comment by chendinh** [comment]
hmm i have no new idea, I got that error too. I just use web3modal and the deeplink work well again. https://docs.walletconnect.com/2.0/web3modal/about
